From 4afeb3c6d23e774251adcccce07d0c440714487f Mon Sep 17 00:00:00 2001
From: Jason Abele <jason@jasonabeleconsulting.com>
Date: Wed, 22 Jul 2015 17:33:59 -0700
Subject: [PATCH 2/2] rtl8723bs: add modules_install and correct depmod

When building an out of tree module in a cross compile environment,
respect the $DEPMOD used with the kernel and add a modules_install make
target.
---
 Makefile | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index e05ef56..28e1ceb 100644
--- a/Makefile
+++ b/Makefile
@@ -7,6 +7,8 @@ EXTRA_CFLAGS += -Wall
 #EXTRA_CFLAGS += -pedantic
 #EXTRA_CFLAGS += -Wshadow -Wpointer-arith -Wcast-qual -Wstrict-prototypes -Wmissing-prototypes
 
+INSTALL_MOD_PATH := $(PREFIX)
+
 EXTRA_CFLAGS += -I$(src)/include -I$(src)/hal -g
 
 ccflags-y += -D__CHECK_ENDIAN__
@@ -176,6 +178,8 @@ MODDESTDIR := /lib/modules/$(KVER)/kernel/drivers/net/wireless/
 INSTALL_PREFIX :=
 endif
 
+MODDESTDIR := $(PREFIX)/lib/modules/$(KERNELRELEASE)/kernel/drivers/net/wireless/
+
 ifneq ($(KERNELRELEASE),)
 
 rtk_core :=	core/rtw_cmd.o \
@@ -217,16 +221,19 @@ all: modules
 modules:
 	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KSRC) M=$(shell pwd)  modules
 
+modules_install:
+	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KSRC) M=$(shell pwd)  INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install
+
 strip:
 	$(CROSS_COMPILE)strip $(MODULE_NAME).ko --strip-unneeded
 
 install:
 	install -p -m 644 $(MODULE_NAME).ko  $(MODDESTDIR)
-	/sbin/depmod -a ${KVER}
+	$(DEPMOD) -a ${KVER}
 
 uninstall:
 	rm -f $(MODDESTDIR)/$(MODULE_NAME).ko
-	/sbin/depmod -a ${KVER}
+	$(DEPMOD) -a ${KVER}
 
 config_r:
 	@echo "make config"
-- 
1.9.1

